### PR TITLE
`plot_slice` with target values other than objective value

### PR DESCRIPTION
**Pull request by not522**
## Motivation
This PR enables flexible plotting in `plot_slice`, same as #2064.
## Description of the changes
Added `target` and `target_name` arguments to `plot_slice`.

### CODE REVIEW

File: optuna/visualization/matplotlib/_slice.py [hunk 1]
```diff
@@ -73,7 +91,10 @@ def _get_slice_plot(study: Study, params: Optional[List[str]] = None) -> "Axes":
     plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
 
     # Prepare data.
-    obj_values = [t.value for t in trials]
+    if target is None:
+        obj_values = [cast(float, t.value) for t in trials]
```
Comment (not522) [inline optuna/visualization/matplotlib/_slice.py:95]: Note: `t.value` is guaranteed not to be `None` because they are filtered by `TrialState.COMPLETE`.

**Comment by crcrpar** [comment]
Kinda general question. What's the expected behavior when `target` is not `None` and `target_name` is the default value?

**Comment by not522** [comment]
If additional arguments are default, the expected behavior is not changed from `master`. Objective value v.s. params are plotted.

**Review by hvy — APPROVED** [review]
LGTM!

@crcrpar, could you jump in as the second reviewer? Let me know if you're busy and I can reassign it.

**Review by crcrpar — APPROVED** [review]
I'd use `get_trials(deepcopy=True, states=(TrialState.COMPLETE,)` in https://github.com/optuna/optuna/pull/2093/files#diff-7abd7081f5c54fa4ae51ec6d2df3c980fef01773ee209731b64ccb6f74867129R82 though this isn't directly related to the goal of this pull request.
Looks good.

**Pull request merged**